homeassistant/components/energy/sensor.py

```diff
@@ -298,20 +313,29 @@ def _update_cost(self) -> None:
                 )
             return
 
-        if reset_detected(
+        if state_class != STATE_CLASS_TOTAL_INCREASING and energy_state.attributes.get(
+            ATTR_LAST_RESET
+        ) != self._last_energy_sensor_state.attributes.get(ATTR_LAST_RESET):
+            # Energy meter was reset, reset cost sensor too
+            energy_state_copy = copy.copy(energy_state)
+            energy_state_copy.state = "0.0"
```

Comment:
Using `copy.copy` is weird. Especially since state objects should be immutable. Why not just create a new state object but with different state?

Reply:
Or maybe we use a namedtuple/dataclass to keep track of the data we want to store ?